run-prtgmpprobe.sh

```diff
@@ -42,19 +42,29 @@ done
 
 # Handling Env vars
 
-# Get/Generate a probe id from PRTGMPPROBE__ID_FILE or from PRTGMPPROBE__ID
-if [ -z "${PRTGMPPROBE__ID-}" ] ; then
-    if [ ! -f "${PRTGMPPROBE__ID_FILE}" ] ; then
-        cat /proc/sys/kernel/random/uuid > ${PRTGMPPROBE__ID_FILE} || (
-            error "Unable to write to ${PRTGMPPROBE__ID_FILE}. Please either set PRTGMPPROBE__ID in the container environment or make sure the location ${PRTGMPPROBE__ID_FILE} is writable."
-            echo >&2 " "
-            echo >&2 "Example:"
-            echo >&2 "PRTGMPPROBE__ID=$(cat /proc/sys/kernel/random/uuid)"
-            exit 1
-        )
+if [ ! -f "${PRTGMPPROBE__CONFIG_FILE}" ] ; then
+    error "Configuration file ${PRTGMPPROBE__CONFIG_FILE} does not exist. Please create one."
+    echo >&2 " "
+    echo >&2 "Example:"
+    ${PRTGMPPROBE__BINARY} example-config >&2
+    exit 1
+fi
+
+# Get/Generate a probe id from PRTGMPPROBE__ID_FILE or from PRTGMPPROBE__ID if not set in PRTGMPPROBE__CONFIG_FILE
+if [ ! grep -q "^id:" "${PRTGMPPROBE__CONFIG_FILE}" ] ; then
+    if [ -z "${PRTGMPPROBE__ID-}" ] ; then
+        if [ ! -f "${PRTGMPPROBE__ID_FILE}" ] ; then
+            cat /proc/sys/kernel/random/uuid > ${PRTGMPPROBE__ID_FILE} || (
+                error "Unable to write to ${PRTGMPPROBE__ID_FILE}. Please either set PRTGMPPROBE__ID in the container environment, 'id:' in the ${PRTGMPPROBE__CONFIG_FILE} or make sure the location ${PRTGMPPROBE__ID_FILE} is writable."
+                echo >&2 " "
+                echo >&2 "Example:"
+                echo >&2 "PRTGMPPROBE__ID=$(cat /proc/sys/kernel/random/uuid)"
```

Comment:
Is this intentional that it already generates and prints a UUID here?
I would have expected this string in single-quotes or with `\$` so that it prints the literal cat... command.

Reply:
no its intentional, so the user directly has one line to use. The same mechanism is used with the `example-config` commands, they generate all necessary values aswell